netket/experimental/sampler/metropolis_pt.py

```diff
@@ -161,16 +182,85 @@ def __init__(self, *args, n_replicas: int = 32, **kwargs):
             machine_pow: The power to which the machine should be exponentiated to generate the pdf (default = 2).
             dtype: The dtype of the states sampled (default = np.float32).
         """
+        # if beta is not provided, linear distribution of 32 temperatures
+        if betas is None:
+            if n_replicas is None:
+                n_replicas = 32
+            beta_distribution = "lin"
+
+        # beta is provided
+        else:
+            # either we know the distribution
+            if isinstance(betas, str):
+                beta_distribution = betas
+                if not (beta_distribution == "lin" or beta_distribution == "log"):
+                    raise ValueError(
+                        f"""To initialize the temperatures with a string, you must choose between "lin" for a linear distribution and "log" for a logarithmic distribution. Instead got "{beta_distribution}"."""
+                    )
+
+                # default if not provided
+                if n_replicas is None:
+                    n_replicas = 32
+
+                # list is empty if we have a string
+                betas = None
+
+            # or we have the list
+            else:
+                betas = jnp.array(betas, dtype=jnp.float64).reshape(-1)
+
+                # verify that the list is compatible with n_replicas
+                if n_replicas is not None and betas.shape[-1] != n_replicas:
+                    raise ValueError(
+                        f"""The dimension of the array beta and the number of replicas must match, instead, got {n_replicas} replicas but {betas.shape[-1]} values for beta."""
+                    )
+
+                # is the list valid
+                if not (jnp.isclose(jnp.max(betas), 1) and jnp.min(betas) > 0):
+                    raise ValueError(
+                        rf"""The values for beta should be in (0,1] and obligatory contain beta=1, instead got [{jnp.min(betas):.2f},{jnp.max(betas):.8f}]."""
+                    )
+
+                # sort and adapt the rest correspondingly
+                betas = jnp.sort(betas, descending=True)  # we need beta[0] = 1
+                n_replicas = betas.shape[-1]
+                beta_distribution = "custom"
```

Comment:
Sorry for rewriting it a bit heavily, but I would reformat the order of checks in this order to make it easier to follow and read.
Otherwise it's good.

Feel free to fix the error messages which I might have ruined a bit.

```suggestion
        if isinstance(betas, str):
            betas = betas.lower()
            if betas not in ["linear", "lin", "logarithmic", "log"]:
                raise ValueError(
                    f"""To initialize the temperatures with a string, you must choose between
                    "lin" for a linear distribution and "log" for a logarithmic distribution.
                    Instead got "{betas}".
                    """
                )
            if n_replicas is None:
                n_replicas = 32

            # truncate to last 3
            beta_distribution = betas[:3]
            betas = None
        elif isinstance(betas, jax.Array) or isinstance(betas, list):
            # TODO: this defaults to float32/64 depending on what is enabled.
            # Might need to think about a better default here.
            betas = jnp.array(betas, dtype=float)
            if betas.ndim != 1:
                raise ValueError("betas must have exactly 1 dimension.")
            if n_replicas is not None:
                raise ValueError(
                    """
                    Cannot specify the list of betas and n_replicas at the same time.
                    The number of replicas will be inferred automatically from the length
                    of the vector of inverse temperatures.
                    """
                )
            # we need beta[0] = 1, so we sort and check that the temperatures are valid
            betas = jnp.sort(betas, descending=True)  
            if not (jnp.isclose(betas[0], 1) and betas[-1] > 0):
                raise ValueError(
                    rf"""The values for beta should be in (0,1] and obligatory contain beta=1, instead got [{jnp.min(betas):.2f},{jnp.max(betas):.8f}]."""
                )

            beta_distribution = "custom"
            n_replicas = betas.shape[-1]
        else:
            raise TypeError("`betas` must be a string or a vector of inverse temperatures.")
```